### ISSUE TITLE
Deny http access to AWS sqs queue 

### ISSUE THREAD

**Issue by ksbde** [aws/karpenter-provider-aws#5901]
### Description

**What problem are you trying to solve?**
After reading the following documents [SQS security best practices](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-security-best-practices.html)
I would like to get ability for overriding AWS SQS access policy or deny access through HTTP to SQS queue due to security reason.
Just for now, all unencrypted request are allowed in karpenter [SQS](https://github.com/aws/karpenter-provider-aws/blob/main/website/content/en/preview/getting-started/getting-started-with-karpenter/cloudformation.yaml#L283-L302) queue.

**How important is this feature to you?**
It's important to be up to date with AWS Security Best Practices.

**Comment by jonathan-innis** [comment]
@ksbde Do you need to be able to directly override the SQS access policy in the cloudformation? The CloudFormation is meant as a "Getting started" template and we generally are expecting folks to use their own IaC representation _based_ off of this initial recommendation. 

TLDR: We're hoping that this CF is a streamlined implementation and doesn't have too many options surfaced within it because this can end up being a more confusing path for people getting started with the project.
